Added instructions for updating Badger

kwsapphire opened the pull request:
Created with help from AboveTheChemist and Derek Icelord. This document aims to provide instructions so others are able to help update the Badger database when new badges are released in City of Heroes Homecoming.

n15g commented:
If possible, can you convert the ODT file to markdown and include it in the `README.md` file, or as a separate .md file? Alternatively, if there's a real need for rich-text formatting, host it in something like Google Sheets and pop a link to it into the readme. 
Open Document files are binary and don't source-control well, as well as requiring downloading to view. Markdown files are readable directly on github.

Also, it may be worth submitting this against the main https://github.com/n15g/badger repo instead, as while this repo does contain the badge data, the UI is where most people will first go looking and the process of updating would apply to any content dbs that get created or forked.

kwsapphire commented:
Hello,I've added the instructions to the website directory, and have asked AboveTheChemist and Derek Icelord to review my markup and ensure I've captured everything correctly. I'll request a merge once they've reviewed.
What do you think of adding the update instructions to the ReadMe file in both the website AND the database directories? 
Thank you,Kelly

    On Thursday, December 12, 2024 at 11:52:46 PM EST, Nicholas Armstrong ***@***.***> wrote:  
 
 


If possible, can you convert the ODT file to markdown and include it in the README.md file, or as a separate .md file? Alternatively, if there's a real need for rich-text formatting, host it in something like Google Sheets and pop a link to it into the readme.
Open Document files are binary and don't source-control well, as well as requiring downloading to view. Markdown files are readable directly on github.

Also, it may be worth submitting this against the main https://github.com/n15g/badger repo instead, as while this repo does contain the badge data, the UI is where most people will first go looking and the process of updating would apply to any content dbs that get created or forked.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

n15g commented:
Not a bad idea, it can’t hurt to have it in both spots!

n15g commented on README.md:
```diff
@@ -16,3 +16,404 @@ const contentDb = new CohContentDb();
+Each indented (block-quoted) line below is an example of what you should include or modify in badge .ts files (each on separate lines). A full example file without notes is included at the end. The individual sections below use different badges as examples so don’t expect consistency until the mock .ts file at the end.
+&nbsp;<BR>
+&nbsp;<BR>
+> import {ALIGNMENT\_ANY, Alternate, BadgePartialType, BadgeType, IBadgeData} from "coh-content-db";
```
Rather than quote blocks, use code blocks for the parts that contain code.

wrap in a single backtick (`) for inline and three backticks (```) for code blocks, like the following:

````
```
import {ALIGNMENT\_ANY, Alternate, BadgePartialType, BadgeType, IBadgeData} from "coh-content-db";
```
````

You also shouldn't need to worry about the BRs or non-breaking spaces then 👍🏻

n15g commented:
Another couple of tweaks, sorry... just to clean up the project structure.
Filenames should be `kebab-case` (all lowercase with dashes between words), except for the README.md file which is uppercase by github's convention.

Also, can you move the example image and the badge grant file into the `docs` and `docs/images` folder to keep the root clear?

kwsapphire commented:
Hello! I think I've correctly updated both ReadMe files. AboveTheChemist said he would review this weekend.
I'm super new to GitHub so I apologize for any annoyance I've caused.
I deleted the badge_grant_all.txt and ExampleNeedsCropped.png from the db-homecoming folder, and pointed the links in that ReadMe to the versions in the badger folder. I hope. I also renamed the files to kebab-case. I'm confused about how deleting in github works though, so I hope I didn't mess up! :D
Kelly L Creighton Aleksic 

    On Tuesday, December 17, 2024 at 10:57:25 PM EST, Nicholas Armstrong ***@***.***> wrote:  
 
 


Another couple of tweaks, sorry... just to clean up the project structure.
Filenames should be kebab-case (all lowercase with dashes between words), except for the README.md file which is uppercase by github's convention.

Also, can you move the example image and the badge grant file into the docs and docs/images folder to keep the root clear?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

n15g commented:
Haha, no worries at all; You're doing fine. I'm overly pedantic about my repo structure, so I'm a pain to deal with sometimes.

I'll let ATC take a look and then we can merge it in.

kwsapphire commented:
Hi there, and happy new year! 
I just copied over my most recent edits from the badger readme to the content readme, and poked AboveTheChemist again to see if he wants to review before I request a merge.
Thank you for your patience. :) 
-Kelly/Sapphire  

    On Friday, December 20, 2024 at 06:21:08 PM EST, Nicholas Armstrong ***@***.***> wrote:  
 
 


Haha, no worries at all; You're doing fine. I'm overly pedantic about my repo structure, so I'm a pain to deal with sometimes.

I'll let ATC take a look and then we can merge it in.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
